includes/Core/Tags/Guards/Tag_Environment_Type_Guard.php

```diff
@@ -31,6 +31,16 @@ public function can_activate() {
 		if ( ! function_exists( 'wp_get_environment_type' ) ) {
 			return true;
 		}
-		return 'production' === wp_get_environment_type();
+
+		$allowed_environments = apply_filters(
+			'googlesitekit_allowed_tag_environment_types',
+			array( 'production' )
+		);
+
+		if ( ! is_array( $allowed_environments ) ) {
+			return false;
+		}
```

Comment:
Let's fallback to `array( 'production' )` if `$allowed_environments` is not array or is an empty array.